arch/riscv/scanner_riscv.c

```diff
@@ -509,6 +498,82 @@ void riscv_check_free_space(dbm_thread *thread_data, uint16_t **write_p,
   }
 }
 
+bool riscv_scanner_deliver_callbacks(dbm_thread *thread_data, mambo_cb_idx cb_id,
+                                     uint16_t **o_read_address, riscv_instruction inst,
+                                     uint16_t **o_write_p, uint32_t **o_data_p,
+                                     int const basic_block, cc_type type,
+                                     bool allow_write, bool *stop) {
+
+  bool replaced = false;
+#ifdef PLUGINS_NEW
+    if (global_data.free_plugin > 0) {
+    uint16_t *write_p = *o_write_p;
+    uint32_t *data_p = *o_data_p;
+    uint16_t *read_address = *o_read_address;
+
+    mambo_cond cond = AL;
```

Comment:
We need to read the condition code for the conditional branches, i.e. RISCV_BEQ, RISCV_BNE, RISCV_BLT, RISCV_BGE, RISCV_BLTU, RISCV_BGEU.

Reply:
On second thought, the Arm instructions for which we set the condition variable are dependent on the state register (i.e. a prior comparison). For CBZ / TBZ, which compare a register themselves, we don't set it. We probably need some new API functionality to capture conditional execution with a condition + 2 input registers, and backport it to also cover Arm CBZ / TBZ.

But that's a job for me. Nevermind this request. just fix the indentation above and I'll merge this.